llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -4442,14 +4441,8 @@ PerformADDCombineWithOperands(SDNode *N, SDValue N0, SDValue N1,
   if (!N0.getNode()->hasOneUse())
     return SDValue();
```

Comment:
Looks like this logic here applied some cost heuristics to `mad` folding. It seems like we don't have very good tests for this, and I'm not sure these heuristics are right, but we should probably be careful about discarding this behavior. If a mul has many uses this transformation could increase register pressure. Is there data/rationale for removing this constraint?

Reply:
Changed to require one use and added test.